libs/@blockprotocol/core/src/service-handler.ts

```diff
@@ -112,11 +156,27 @@ export abstract class ServiceHandler {
   ) {
     this.checkIfDestroyed();
 
-    this.coreHandler.registerCallback({
-      callback,
-      messageName,
-      serviceName: this.serviceName,
-    });
+    this.coreQueue.push((coreHandler) =>
+      coreHandler.registerCallback({
+        callback,
+        messageName,
+        serviceName: this.serviceName,
+      }),
+    );
+
+    this.processCoreCallbacks();
```

Comment:
I think it's a little confusing that we have `registerCallback` and `processCoreCallbacks` and they refer to completely different things – maybe this could be `processCoreQueue` or `processCoreActions` or something else?